### PR TITLE
First parser

### PR DESCRIPTION
**Pull request by arsenal9971**
First parser

Ref #12

### CODE REVIEW

File: crawler/parser.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+"""
+	Parser for URL or HTML file"
+"""
+
+from urllib.request import urlopen
+from bs4 import BeautifulSoup
+from urllib.error import HTTPError
```
Comment (lwm) [inline crawler/parser.py]: As I mentioned in the chat, I think we can use the [Selector][1] from `scrapy`.

[1]: http://doc.scrapy.org/en/latest/topics/selectors.html

File: crawler/parser.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+"""
```
Comment (lwm) [inline crawler/parser.py]: Would be nice to see a test for this to better understand it. You can use the `response` fixture in the `conftest.py`.

**Comment by lwm** [comment]
What do you think about using the [readability parser API][1] straight out of the box. It looks simple and can be a good first start?

[1]: https://www.readability.com/developers/api/parser

File: crawler/parser.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+from urllib.request import urlopen
+from bs4 import BeautifulSoup
+from urllib.error import HTTPError
+
+#url = "http://appliedmythology.blogspot.de/"
+# PATH_TO_FILE="/Users/hector/Downloads/test.html"
+
+class ArticleParser():
+
+	def parser_get(link1,type1):
```
Comment (lwm) [inline crawler/parser.py]: This code can go into the `parse_item` function in `pipelines.py` It is defined as a pipeline in `settings.py`

**Comment by lwm** [comment]
Sorry, merged some stuff, so you will also need to rebase on master.

**Comment by arsenal9971** [comment]
Sorry dude, been out of github for a day. Now I will retake this, I will rebase on master, and will solve the issues with the pull request, I am checking now on selector of scrapy and the Readability api. I am not sure the last one will work since with a lot of webpages, I my self tried to use it in my chrome and didnt work for a lot of pages.

**Comment by oliversauter** [comment]
@arsenal9971 I think readability is not really viable, since it is only free for non commercial use. So there might be a problem. Also because of the sheer mass of articles we have to parse, like you already noticed. 

Another parser you can check out is [boilerpipe](https://github.com/kohlschutter/boilerpipe) or some other alternatives(https://www.quora.com/Whats-the-best-method-to-extract-article-text-from-HTML-documents)

**Comment by oliversauter** [comment]
@arsenal9971 on the other hand I just found out that readability was once open source and there are some version on Github: https://github.com/search?utf8=%E2%9C%93&q=readability&type=Repositories&ref=searchresults

This one seems the last active version: https://github.com/luin/readability

**Comment by oliversauter** [comment]
@arsenal9971 I also found the Boilerpipe web api, where you can test the pages:
http://boilerpipe-web.appspot.com/

This is the result for the appliedmythodology.de-article http://boilerpipe-web.appspot.com/extract?url=http%3A%2F%2Fappliedmythology.blogspot.de%2F2016%2F06%2Fthe-non-gmo-food-label-is-lie.html&extractor=LargestContentExtractor&output=htmlFragment&extractImages=&token=

works pretty well.

We might can think of running newspaper3k and boilerpipe...boilerpipe for the text and newspaper3k for the rest (tags, author, title etc). I don't know how precise beautifulSoup is for that. (But we NEED all this information. title and text is not enough)

It seems like we would have to do quite some work to get beautiful soup work reliably for us, falling back on existing, proven solutions is the better option. 
What is your stance on the work that has to be done to get BeautifulSoup running reliably?

**Comment by lwm** [comment]
> Sorry dude, been out of github for a day.

No worries! 

> This one seems the last active version: https://github.com/luin/readability

Something to look into for sure! I like their parsed format. We can build from it and add our own.

> we would have to do quite some work to get beautiful soup work reliably for us

Let's hack this parser into shape tomorrow. The server is waiting on it :)

**Comment by oliversauter** [comment]
> Let's hack this parser into shape tomorrow. The server is waiting on it :)

Good idea, once we have the crawler pipeline working I can go on hunting for good sources and let it run while we get the other stuff running.

**Comment by lwm** [comment]
For reference later: https://readability-python-library.readthedocs.io/en/latest/parser.html

**Comment by lwm** [comment]
More references: https://github.com/buriy/python-readability

**Comment by arsenal9971** [comment]
@lwm @oliversauter  I already solve the problems of format of code in the original parrser that I did on Thursday, I asked for pull request https://github.com/WorldBrain/crawler/compare/master...arsenal9971:master

**Comment by oliversauter** [comment]
About what format are we (you) talking?

I am talking about the output format as a JSON and the respective JSON elements that we need. (title, body, tags, authors, publish_date etc etc)

**Comment by arsenal9971** [comment]
Yes, I will work on that in a moment, in the hacknight, but I just wanted to check why it fails of format of the code I used pylama to solve it, to work over that. I still having one problem "E501 line too long (80 > 79 characters) [pep8]", @lwm any Idea how to solve this, maybe make the names shorter

**Comment by arsenal9971** [comment]
Ok last pull request on this, I finally solve that problem https://github.com/WorldBrain/crawler/compare/master...arsenal9971:master , pleas accept I want to see how does it works the json can be created easly with that function but in another module

**Comment by oliversauter** [comment]
lets talk about it later :)

**Comment by lwm** [comment]
We renamed the repo, so I suggest you fork again the project and re submit this PR!

**Pull request closed without merging**